### PR TITLE
Implemented new time handling for time_map_async

### PR DESCRIPTION
**Pull request by arnasbr**
This PR improves how time is handled for time-map requests

### CODE REVIEW

File: traveltimepy/dto/common.py [hunk 1]
```diff
@@ -140,3 +140,18 @@ class LevelOfDetail(BaseModel):
     scale_type: Literal["simple", "simple_numeric", "coarse_grid"] = "simple"
     level: Optional[Union[int, str]]
     square_size: Optional[int]
+
+
+class TimeInfo:
+    def __init__(self, time_value: datetime):
+        if not isinstance(time_value, datetime):
+            raise ValueError("time must be a datetime object")
+        self.time = time_value
```
Comment (danielnaumau) [inline traveltimepy/dto/common.py:149]: it can be value

File: traveltimepy/sdk.py [hunk 2]
```diff
@@ -232,14 +243,22 @@ async def postcodes_async(
         properties: Optional[List[Property]] = None,
         range: Optional[FullRange] = None,
     ) -> List[PostcodesResult]:
+        if departure_time and arrival_time:
+            raise ApiError("arrival_time and departure_time cannot be both specified")
+
+        time_info = None
```
Comment (danielnaumau) [inline traveltimepy/sdk.py:249]: move it to a function

Comment (danielnaumau) [reply traveltimepy/sdk.py:249]: right now it's a huge duplication

**Review by danielnaumau — CHANGES_REQUESTED** [review]

File: traveltimepy/dto/common.py (hunk 1 above)
Comment (arnasbr) [reply traveltimepy/dto/common.py:147]: I don't think I can avoid using the __init__ method completely if that's what you mean. But simplified it a bit

File: traveltimepy/sdk.py (hunk 2 above)
Comment (arnasbr) [reply traveltimepy/sdk.py:247]: done

File: traveltimepy/sdk.py [hunk 3]
```diff
@@ -124,8 +135,7 @@ async def time_filter_async(
                 search_ids,
                 transportation,
                 properties,
-                departure_time,
-                arrival_time,
+                time_info,
```
Comment (danielnaumau) [inline traveltimepy/sdk.py:138]: get_time_info(). Move everything to a function.

**Review by danielnaumau — CHANGES_REQUESTED** [review]

**Comment by Donatas-L** [comment]
Is this PR stuck or just waiting for reviews? @arnasbr

**Comment by arnasbr** [comment]
> Is this PR stuck or just waiting for reviews? @arnasbr 

@Donatas-L This is a bit of a confusing one. To fix the problem I created a lot of code duplication, and fixing the code duplication reintroduced the original problem. We thought that we're gonna make some breaking changes in the future and fix it properly, but this can just be merged IMO

**Comment by danielnaumau** [comment]
Yes, I'd say that we can merge it

**Review by danielnaumau — APPROVED** [review]

**Pull request merged**